src/item.cpp

```diff
@@ -6143,7 +6179,7 @@ bool item::is_bionic() const
 
 bool item::is_magazine() const
 {
-    return !!type->magazine;
+    return !!type->magazine || contents.has_pocket_type( item_pocket::pocket_type::MAGAZINE );
```

Comment:
This change means that things that use this to gate accessing `type->magazine` will no longer work, and will crash on attempting to access non-existent magazine data.

Reply:
this is true, and i have come across these that do cause crashes and fixed them. islot_magazine is on the chopping block in favor of pocket_data (need to check with yutna for sure but i'm confident that this is true), and if i do not have this bool in here it will backpedal a lot of the work on this pr

Reply:
Not this one
https://github.com/CleverRaven/Cataclysm-DDA/pull/40399/files#diff-f7ca61cf749eba039da961e8a7be1630L8408
`if( !is_magazine() || !type->magazine->protects_contents ) {`